docs/variant_annotation.md

```diff
@@ -0,0 +1,140 @@
+# Annotation Support in Variant Transforms
+
+## Overview
+
+Tha annotation support in Variant Transforms (VT) has two goals:
+* First, we want to be able to properly parse annotation fields in input VCFs.
+* The second goal is to be able to annotate VCF files during the import process,
+using open annotation tools/databases like
+[Variant Effect Predictor](http://ensembl.org/info/docs/tools/vep/index.html)
+(VEP).
+
+## Parsing Annotation Fields
+
+The annotation field format that VT currently supports is based on
+[this spec](http://snpeff.sourceforge.net/VCFannotationformat_v1.0.pdf)
+developed in the SnpEff project. VEP uses the same format and here is a trimmed
+example from a file annotated by VEP (taken from the
+[gnomAD](http://gnomad.broadinstitute.org/) dataset):
+
+VCF Header:
+```
+##INFO=<ID=CSQ,Number=.,Type=String,Description="Consequence annotations from Ensembl VEP. Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|ALLELE_NUM|...>
+```
+Corresponding annotations for a sample variant record where `REF=AAC` and
+`ALT=TAC,C`:
+```
+CSQ=-|downstream_gene_variant|MODIFIER|WASH7P|ENSG00000227232|Transcript|ENST00000423562|unprocessed_pseudogene|||1,...,T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000518655|transcribed_unprocessed_pseudogene|||2
+```
+For each variant record, multiple annotation lists are provided which are
+separated by comma. The first field of each set refers to an alternate allele or
+`ALT` (for details of the format see
+[the spec](http://snpeff.sourceforge.net/VCFannotationformat_v1.0.pdf)).
+Note that the actual record of the above example, has more fields and also each
+alternate allele can be repeated multiple times (with different annotations);
+here we have only selected a small sub-set, one for each `ALT`.
+
+When [`--annotation_fields`](https://github.com/googlegenomics/gcp-variant-transforms/blob/c4659bba2cf577d64f15db5cd9f477d9ea2b51b0/gcp_variant_transforms/options/variant_transform_options.py#L156)
+flag of VT is used, annotation lists are split and moved under
+their corresponding `alternate_bases` records. In other words, there will be a
+repeated record field `CSQ` under each `alternate_bases`:
+
+![A sample CSQ schema](images/CSQ_schema.png)
+
+### Alternate Allele Matching
+
+As it can be seen in the above example, the alternate alleles listed in the
+`CSQ` field are different from `ALT` strings of the variant record (for example
+the `AAC->C` variant is represented by a single `-` which means a deletion).
+This is because, the gnomAD set is run through VEP with the
+[`--minimal`](http://useast.ensembl.org/info/docs/tools/vep/script/vep_options.html#opt_minimal)
+option. VT can recognize this, in fact, currently there are three modes of `ALT`
```

Comment:
nit: this is a bit hard to read; consider rephrasing with two separate sentences (e.g. something like "VT supports matching alt fields in minimal mode. There are actually three modes of ALT matching supported by VT:").

Reply:
Done.